File: C/Algorithms/657.c
#include <stdbool.h>

bool judgeCircle(char *moves) {
  int u = 0, d = 0, l = 0, r = 0;
  while (*moves) {
    switch (*moves++) {
    case 'L':
      l++;
      break;
    case 'R':
      r++;
      break;
    case 'U':
      u++;
      break;
    case 'D':
      d++;
      break;
    }
  }
  return (u == d) && (l == r);
}

File: Javascript/Algorithms/657.js
/**
 * @param {string} moves
 * @return {boolean}
 */
let judgeCircle = moves => {
    let u = d = l = r = 0;
    for (let c of moves){
        switch(c){
            case 'L': l++; break;
            case 'R': r++; break;
            case 'U': u++; break;
            case 'D': d++; break;
        }
    }
    return (u == d) && (l == r);
};
